pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -613,11 +614,11 @@ func (c *Controller) getProxyServiceInstancesByPod(pod *v1.Pod, service *v1.Serv
 			log.Warnf("Failed to find port for service %s/%s: %v", service.Namespace, service.Name, err)
 			continue
 		}
-		// There is only one IP for kube registry
-		proxyIP := proxy.IPAddresses[0]
-
-		out = append(out, c.getEndpoints(proxyIP, int32(portNum), svcPort, svc))
 
+		// there are multi-ip scenarios, such as cni or multus-cni
+		for _, ip := range proxy.IPAddresses {
+			out = append(out, c.getEndpoints(ip, int32(portNum), svcPort, svc))
+		}
```

Comment:
I would rather appending the instances by changing the NetworkEndpoint.Address

Reply:
Already modified